Move from Make to Task

gene-redpanda opened the pull request:
Updates various clients and tooling in the repo and moves us to task from make to bring the repo into alignment with the k8s operator.

gene-redpanda committed:
chore: bump client dockerfile versions

gene-redpanda committed:
chore: update grafana role

gene-redpanda committed:
feat: move to task from make

gene-redpanda committed:
ci: pipeline to use taskfile

gene-redpanda committed:
ci: yamllint config

gene-redpanda committed:
chore: delete makefile

chrisseto commented on .tasks/connect.yml:
```diff
@@ -0,0 +1,150 @@
+
+vars:
+  ARTIFACT_DIR: "{{.PWD}}/artifacts"
+  DEPLOYMENT_ID: '{{.DEPLOYMENT_ID | default "devex-cicd"}}'
```
I see this defined and some other variables with the same definitions multiple times (and sometimes not used). Is that intentional? If not would it be better to move share variables into Taskfile.yml at the top level?

chrisseto commented on .tasks/tools.yml:
```diff
@@ -0,0 +1,129 @@
+      - mkdir -p {{.ARTIFACT_DIR}}/tmp
+      - mkdir -p {{.ARTIFACT_DIR}}/bin
+      - |
+        OS=$(uname)
```
Taskfile has a way to perform this check itself. I don't think I've used it but it may be a bit nicer?

chrisseto commented on .tasks/tools.yml:
```diff
@@ -0,0 +1,129 @@
+      - |
+        OS=$(uname)
+        if [ "$OS" = "Darwin" ]; then
+          curl -L {{.MAC_RPK}} -o {{.ARTIFACT_DIR}}/tmp/rpk.zip
```
Just my two cents, I find it better to put non-local variables into the scope of the task so it's clear what is local to this task and what's expected to be shared.

gene-redpanda committed:
test

gene-redpanda committed:
fix: ensures no ssh regen

When run during the connect test without this particular method of testing for the existence of keys the connect test fails.

gene-redpanda committed:
ci: move citsubus to new concurrency group to reduce test time
